### PR TITLE
Add KArchive to client Dockerfiles

### PR DESCRIPTION
**Pull request by claucambra**
This should fix failures with this PR on the desktop client side: https://github.com/nextcloud/desktop/pull/4768

### CODE REVIEW

File: client/Dockerfile [hunk 1]
```diff
@@ -182,6 +182,31 @@ RUN if [ "$BUILD_QT" != "1" ] ; then \
...
+    cd build && \
+    cmake -G Ninja -DCMAKE_BUILD_TYPE=Release .. && \
+    ninja install && \
+    cd .. && \
+    rm -rf extra-cmake-modules \
+    ; fi
+    
+RUN if [ "$BUILD_QT" != "1" ] ; then \
+    cd /tmp && \
+    git clone https://invent.kde.org/frameworks/karchive.git -b v5.96.0 && \
```
Comment (mgallien) [inline client/Dockerfile:200]: add `--depth 1` to make a swallow clone

File: client-appimage/Dockerfile [hunk 2]
```diff
@@ -171,3 +171,28 @@ RUN if [ "$BUILD_QT" != "1" ] ; then \
     cd .. && \
     rm -rf qtkeychain \
     ; fi
+
+# Install KArchive
+RUN if [ "$BUILD_QT" != "1" ] ; then \
+    cd /tmp && \
+    git clone https://invent.kde.org/frameworks/extra-cmake-modules.git -b v5.96.0 && \
```
Comment (mgallien) [inline client-appimage/Dockerfile:178]: same comment

File: client-appimage/Dockerfile [hunk 3]
```diff
@@ -171,3 +171,28 @@ RUN if [ "$BUILD_QT" != "1" ] ; then \
...
+    cd build && \
+    cmake -G Ninja -DCMAKE_BUILD_TYPE=Release .. && \
+    ninja install && \
+    cd .. && \
+    rm -rf extra-cmake-modules \
+    ; fi
+
+RUN if [ "$BUILD_QT" != "1" ] ; then \
+    cd /tmp && \
+    git clone https://invent.kde.org/frameworks/karchive.git -b v5.96.0 && \
```
Comment (mgallien) [inline client-appimage/Dockerfile:190]: same comment

File: client/Dockerfile (hunk 1 above)
Comment (claucambra) [reply client/Dockerfile]: Okay, added `--depth 1`

**Review by mgallien — APPROVED** [review]

**Review by mgallien — CHANGES_REQUESTED** [review]
build of ecm framework is failing (speaks about missing sphinx module for qt docs and I cannot remember if that one is really mandatory or we could just disable it)

**Comment by claucambra** [comment]
> build of ecm framework is failing (speaks about missing sphinx module for qt docs and I cannot remember if that one is really mandatory or we could just disable it)

Fixed the build :)

**Review by mgallien — APPROVED** [review]

**Pull request merged**